Sources/Architecture/Coordinator.swift

```diff
@@ -163,19 +163,18 @@ public protocol CoordinatorDelegate: class {
 
 /// Used typically on view controllers to refer to it's coordinator
 public protocol Coordinated {
+    associatedtype C: Coordinator
+    var coordinator: C! { get set }
     func getCoordinator() -> Coordinator?
     func setCoordinator(_ coordinator: Coordinator)
 }
 
-public class CoordinatorSegue: UIStoryboardSegue {
-
-    open var sender: AnyObject!
-
-    override public func perform() {
-        guard let source = self.source as? Coordinated else {
-            return
-        }
+public extension Coordinated {
+    func getCoordinator() -> Coordinator? {
+        return coordinator
+    }
 
-        source.getCoordinator()?.navigate(from: self.source, to: destination, with: identifier, and: sender)
+    mutating func setCoordinator(_ coordinator: Coordinator) {
+        self.coordinator = coordinator as? C
```

Comment:
I even think it'd never crash because `associatedtype C: Coordinator`, `coordinator: Coordinator` so `coordinator as! C` will always succeed.